src/Mentat.Repository/Services/CardService.cs

```diff
@@ -35,9 +30,38 @@ public void RemoveCard(string id)
             _cards.DeleteOne(card => card.Id.Equals(id));
         }
 
-        public void UpdateCard(string id, Card card)
+        public void SaveCard(string id, IFormCollection collection)
+        {
+            var card = new Card();
+            if (id != null)
+            {
+                card = GetCard(id);
+                if (card == null)
+                {
+                    throw new Exception("Card ID invalid.");
+                }
+            }
+
+            card.Notes = collection["notes"];
+            card.Subject = collection["subject"];
+            card.Question = collection["question"];
+            card.Answer = collection["answer"];
+            card.DifficultyLevel = collection["difficultyLevel"];
+
+            InsertOrUpdateCard(card);
+        }
+
+        private void InsertOrUpdateCard(Card card)
```

Comment:
While I see that this works, it violates the Single-Responsibility principle of Software Engineering. We should have a function for insert and a separate one for delete.

Reply:
It does indeed violate Single-Responsibility principle, though it is rather convenient. It might be the case that Mongo has native API to do this exact thing; I know Django ORM can decide to insert or update based on the parameter list and some other clues. I would definitely check the Mongo API to see if the repository service could leverage a native implementation that would allow you to get the same result while still adhering to Single-Responsibility in our code.

Reply:
@dyohn, I'm able to do the upsert with just the one line:
_cards.ReplaceOne(c => c.Id.Equals(card.Id), card, new ReplaceOptions { IsUpsert = true });

Reply:
Removed InsertOrUpdateCard and replaced with the line above.
Changes committed. Resolving.